File: plugins/sqldelight/src/main/java/eu/geekhome/sqldelightplugin/SqlDelightPlugin.java
package eu.geekhome.sqldelightplugin;

import com.geekhome.common.extensibility.PluginMetadata;
import com.geekhome.common.localization.Resource;

import org.pf4j.PluginWrapper;
import org.pf4j.Plugin;

public class SqlDelightPlugin extends Plugin implements PluginMetadata {

    public SqlDelightPlugin(PluginWrapper wrapper) {
        super(wrapper);
    }

    @Override
    public void start() {
        System.out.println("SQLDelight plugin.start()");
    }

    @Override
    public void stop() {
        System.out.println("SQLDelight plugin.stop()");
    }

    @Override
    public Resource getName() {
        return R.plugin_name;
    }

    @Override
    public Resource getDescription() {
        return R.plugin_description;
    }
}

File: plugins/sqldelight/src/main/java/eu/geekhome/sqldelightplugin/SqlDelightPlugin.kt
package eu.geekhome.sqldelightplugin

import eu.geekhome.domain.extensibility.PluginMetadata
import eu.geekhome.domain.localization.Resource
import org.pf4j.Plugin
import org.pf4j.PluginWrapper

class SqlDelightPlugin(wrapper: PluginWrapper?) : Plugin(wrapper), PluginMetadata {
    override fun start() {
        println("Starting SQLDelight plugin")
    }

    override fun stop() {
        println("SQLDelight plugin.stop()")
    }

    override val name: Resource = R.plugin_name
    override val description: Resource = R.plugin_description
}
